share/spack/setup-env.sh

```diff
@@ -229,7 +221,7 @@ if [ "${need_module}" = "yes" ]; then
         #activate it!
         export MODULE_PREFIX=${module_prefix}
         _spack_pathadd PATH "${MODULE_PREFIX}/Modules/bin"
-        module() { eval `${MODULE_PREFIX}/Modules/bin/modulecmd ${SPACK_SHELL} $*`; }
+        module() { eval $(${MODULE_PREFIX}/Modules/bin/modulecmd $(basename ${SHELL}) $*); }
```

Comment:
I think `$(...)` is a Bash-ism, it isn't POSIX-compatible. I guess we'll need a variable, but we don't need to export it or check if it already exists.